### PR TITLE
Russell-GCAM3

### PR DESCRIPTION
**Pull request by russellhz**
#636

### CODE REVIEW

File: R/zchunk_L102.GDP.R [hunk 1]
```diff
@@ -285,6 +299,100 @@ module_socioeconomics_L102.GDP <- function(command, ...) {
...
+    # rebasing GDP to 2010 USD at country level
+    # Calculate the GDP ratios from the first year in the projections. Use this ratio to project GDP from historical dataset in final historical period
+    gdpRatio_GCAM3_ctry_Yfut <- gdp_mil90usd_GCAM3_ctry_Y %>%
+      group_by(iso) %>%
+      mutate(base = value[year == socioeconomics.FINAL_HIST_YEAR]) %>%
+      ungroup() %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      transmute(iso, year, ratio = value / base)
+
+    #Use these ratios to build the GDP trajectories by old GCAM3
```
Comment (bpbond) [inline R/zchunk_L102.GDP.R]: `# Use`

File: R/zchunk_L242.Bld_Inc_Elas_scenarios.R [hunk 2]
```diff
@@ -73,6 +78,20 @@ module_socioeconomics_L242.Bld_Inc_Elas_scenarios <- function(command, ...) {
...
                          "L102.pcgdp_thous90USD_Scen_R_Y") })
 
+    # L242.IncomeElasticity_bld_GCAM3: building sector income elasticity for GCAM 3.0 socioeconomics
+    # For the GCAM 3.0 scenario, calculate the per-capita GDP
+    L242.IncomeElasticity_bld_GCAM3 <- L102.gdp_mil90usd_GCAM3_R_Y %>%
+      left_join_error_no_match(L101.Pop_thous_GCAM3_R_Y, by = c("GCAM_region_ID", "year")) %>%
+      filter(year %in% FUTURE_YEARS) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      transmute(region, year, pcgdp_90thousUSD = value.x / value.y) %>%
+      mutate(income.elasticity = approx(x = A42.inc_elas$pcgdp_90thousUSD, y = A42.inc_elas$inc_elas,
```
Comment (bpbond) [inline R/zchunk_L242.Bld_Inc_Elas_scenarios.R]: Why using `approx` here? Can we tweak `approx_fun` to let you use that?

**Comment by bpbond** [comment]
@pralitp Do you mind reviewing, since Haewon is away I think?

File: R/zchunk_L242.Bld_Inc_Elas_scenarios.R (hunk 2 above)
Comment (russellhz) [inline R/zchunk_L242.Bld_Inc_Elas_scenarios.R]: We are basing the interpolation on an assumption file. To use approx_fun, I think I would need to bind the rows of the assumption files, probably after doing a `repeat_add_columns` for all regions, then `approx_fun`, then remove the assumption file data? Seems messy, but let me know if you have a cleaner method. Maybe adding all unique `pcgdp_90thousUSD` to the assumption file then doing approx_fun there would work.

This is also done in the file previously and in L252

Comment (bpbond) [inline R/zchunk_L242.Bld_Inc_Elas_scenarios.R]: Ugh, does seem messy. OK maybe just add a comment explaining better why we're doing it this way. Thanks.

**Comment by bpbond** [comment]
Holding on #656

**Comment by bpbond** [comment]
Runs fine on my system, so merging even though Travis is borked ( #656 ).

**Pull request merged**